dbt/adapters/sqlserver/connections.py

```diff
@@ -1,36 +1,21 @@
-from contextlib import contextmanager
-
-import pyodbc
 import os
-import time
 import struct
+import time
+from contextlib import contextmanager
+from dataclasses import dataclass
+from itertools import chain, repeat
+from typing import Callable, Mapping, Optional
 
 import dbt.exceptions
+import pyodbc
 from dbt.adapters.base import Credentials
 from dbt.adapters.sql import SQLConnectionManager
-from azure.identity import DefaultAzureCredential
-
 from dbt.logger import GLOBAL_LOGGER as logger
-
-from dataclasses import dataclass
-from typing import Optional
+from azure.core.credentials import AccessToken
+from azure.identity import AzureCliCredential, DefaultAzureCredential
```

Comment:
any reason these imports got re-arranged? no strong opinion on this side... I believe I used VSCode's "sort imports" context menu

Reply:
I'll put them back in the same order and open a separate PR for this. I do not care too much either. The following resources are useful:

According to [PEP8](https://www.python.org/dev/peps/pep-0008/#imports), Python imports should be on separate lines (per library) and the imports are grouped in three blocks:
1. Standard library imports.
2. Related third party imports.
3. Local application/library specific imports.

Next to that, it is common to sort the imports within on of these three group alphabetically - to improve searchability. Finally, [some](https://stackoverflow.com/questions/20762662/whats-the-correct-way-to-sort-python-import-x-and-from-x-import-y-statement) put the `import ...` before `from ... import ...`, or vice versa or mix them.